operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,248 @@
+  auto point_end = point_start + line_length * stride;
+
+  for (int64_t i = 0; i < thickness; ++i) {
+    std::copy_n(color.begin(), color.size(), point_start + i * image.channels);
+  }
+  for (auto xy_point_start = point_start + stride; xy_point_start < point_end; xy_point_start += stride) {
+    std::copy_n(point_start, thickness * image.channels, xy_point_start);
+  }
+}
+
+// Draw a box on the image with given thickness and color
+void DrawBox(ImageView& image, gsl::span<const float> box, BoundingBoxFormat bbox_mode,
+             gsl::span<const uint8_t> color, int64_t thickness) {
+  // -------(1)------
+  // |              |
+  //(2)            (4)
+  // |              |
+  // -------(3)------
+  float point_x1 = box[0];
+  float point_x2 = box[1];
+  float point_x3 = box[2];
+  float point_x4 = box[3];
+  if (bbox_mode == BoundingBoxFormat::Center_XYWH) {
+    point_x1 = box[0] - point_x3 / 2;
+    point_x2 = box[1] - point_x4 / 2;
+    point_x3 = box[0] + point_x3 / 2;
+    point_x4 = box[1] + point_x4 / 2;
+  } else if (bbox_mode == BoundingBoxFormat::XYWH) {
+    point_x1 = box[0];
+    point_x2 = box[1];
+    point_x3 = box[0] + box[2];
+    point_x4 = box[1] + box[3];
+  }
+  int64_t x_start = static_cast<int64_t>(std::rintf(point_x1));
+  int64_t y_start = static_cast<int64_t>(std::rintf(point_x2));
+  int64_t x_end = static_cast<int64_t>(std::rintf(point_x3));
+  int64_t y_end = static_cast<int64_t>(std::rintf(point_x4));
+
+  auto offset = thickness / 2;
+  x_start -= offset;
+  y_start -= offset;
+  x_end += (thickness - offset);
+  y_end += (thickness - offset);
+
+  x_start = std::max<int64_t>(x_start, 0L);
+  y_start = std::max<int64_t>(y_start, 0L);
+  x_end = std::min<int64_t>(x_end, image.height - 1);
+  y_end = std::min<int64_t>(y_end, image.width - 1);
+
+  thickness = std::clamp<int64_t>(thickness, 1, std::min(x_end - x_start, y_end - y_start));
+  if (x_end - x_start < thickness || y_end - y_start < thickness) {
+    // "invalid box, It's invalid to draw a point to represent a box.";
+    return;
+  }
+
+  if (x_start + thickness >= image.height || y_start + thickness >= image.width) {
+    // "invalid box, It's invalid to draw a point to represent a box.";
+    return;
+  }
+
+  // line  (1) --
+  DrawLineInHorizon(image, x_start, y_start, y_end - y_start, color, thickness);
+
+  // line  (2) |--
+  DrawLineInVertical(image, x_start, y_start, x_end - x_start, color, thickness);
+
+  // line  (3) __
+  DrawLineInHorizon(image, x_end - thickness, y_start, y_end - y_start, color, thickness);
+
+  // line  (4) --|
+  DrawLineInVertical(image, x_start, y_end - thickness, x_end - x_start, color, thickness);
+}
+
+void DrawBoxesForNumClasses(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  std::unordered_map<float, size_t> color_used;
+
+  for (size_t i = 0; i < boxes.NumBoxes(); ++i) {
+    const auto box = boxes.GetBox(i);
+    if (color_used.find(box[5]) == color_used.end()) {
```

Comment:
we should have a constant providing context instead of directly using the value '5'